### DISCUSSION TITLE
How to Prevent permissions being requested on Android devices?

### DISCUSSION THREAD

**Discussion by tremzo** [ant-media/Ant-Media-Server#5722] [FAQs]
Due to the design of our application, we have to ask users for these permissions beforehand, informing them that if the browser requests the permissions to accept them. 

Although we managed to request the permissions before, on Android devices we get the message "Camera or Mic is being used by some other process that does not allow these devices to be read." which is in the `AntMedia.js` file and it doesn't let us continue with the correct functioning of the app.

Is there a way to prevent the permissions requested by "AntMedia" from being executed? Because we believe that the error may arise because the permissions are requested twice in a row.

**Comment by lastpeony** [comment]
Go to WebRTCClient.java in android SDK
https://github.com/ant-media/WebRTC-Android-SDK/blob/master/webrtc-android-framework/src/main/java/io/antmedia/webrtcandroidframework/WebRTCClient.java#L765C13-L765C29
Please review the "checkPermissions" method.





This method checks permissions and if they are not given it asks for it. If permissions are given it proceeds with init.

**Reply by tremzo** [reply]
@lastpeony how does this work on mobile web? So if you do not want the permissions to be asked on mobile web?

**Reply by lastpeony** [reply]
MediaDevices .getUserMedia() asks for permission on javascript SDK
Have a look at getMedia function of media_manager.js
https://github.com/ant-media/StreamApp/blob/6829b3b6f8627b7dc9513ac387be627ca435d88f/src/main/webapp/js/media_manager.js#L582

Its called by openStream function
https://github.com/ant-media/StreamApp/blob/6829b3b6f8627b7dc9513ac387be627ca435d88f/src/main/webapp/js/media_manager.js#L582
and its called by initLocalStream which is called by WebRTCAdaptor 
https://github.com/ant-media/StreamApp/blob/6829b3b6f8627b7dc9513ac387be627ca435d88f/src/main/webapp/js/media_manager.js#L233

initialize
https://github.com/ant-media/StreamApp/blob/6829b3b6f8627b7dc9513ac387be627ca435d88f/src/main/webapp/js/webrtc_adaptor.js#L430
